### ISSUE TITLE
ConcurrentModificationException in EC2FleetCloud.updateStatus

### ISSUE THREAD

**Issue by jzila** [jenkinsci/ec2-fleet-plugin#79]
```
May 29, 2019 7:00:30 PM INFO com.amazon.jenkins.ec2fleet.EC2FleetCloud updateStatus
Fleet (linux) no longer has the instance i-0fe511f1b6c241ea3, removing from Jenkins.
May 29, 2019 7:00:30 PM SEVERE hudson.triggers.SafeTimerTask run
Timer task com.amazon.jenkins.ec2fleet.CloudNanny@713204b8 failed
java.util.ConcurrentModificationException
	at java.util.HashMap$HashIterator.nextNode(HashMap.java:1445)
	at java.util.HashMap$KeyIterator.next(HashMap.java:1469)
	at com.amazon.jenkins.ec2fleet.EC2FleetCloud.updateStatus(EC2FleetCloud.java:379)
	at com.amazon.jenkins.ec2fleet.CloudNanny$1.call(CloudNanny.java:48)
	at com.amazon.jenkins.ec2fleet.CloudNanny$1.call(CloudNanny.java:44)
	at hudson.model.Queue._withLock(Queue.java:1438)
	at hudson.model.Queue.withLock(Queue.java:1299)
	at com.amazon.jenkins.ec2fleet.CloudNanny.doRun(CloudNanny.java:44)
	at hudson.triggers.SafeTimerTask.run(SafeTimerTask.java:72)
	at jenkins.security.ImpersonatingScheduledExecutorService$1.run(ImpersonatingScheduledExecutorService.java:58)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by jzila** [comment]
However this might just be because I had to rollback to 1.21 due to #78.

**Comment by terma** [comment]
@jzila NPE #78 fixed in 1.3.1 please comment this one if you still see this problem with fixed NPE, thx a lot

**Comment by jzila** [comment]
It appears that this issue is gone. Thanks!
